src/components/menu/menu.spec.ts

```diff
@@ -354,28 +363,82 @@ describe('MenuComponent', () => {
         fixture.detectChanges();
         await fixture.whenStable();
 
-        spyOn(component, 'item1Activated');
+        spyOn(component, 'onActivate');
 
         const item1Element = overlayContainerElement.querySelector('#menu-item-1') as HTMLButtonElement;
         expect(item1Element).toBeTruthy();
         item1Element.click();
 
-        expect(component.item1Activated).toHaveBeenCalledTimes(1);
+        expect(component.onActivate).toHaveBeenCalledTimes(1);
     });
 
     it('should emit an activated event when pressing enter key on a menu item', async () => {
         component.trigger.openMenu();
         fixture.detectChanges();
         await fixture.whenStable();
 
-        spyOn(component, 'item1Activated');
+        spyOn(component, 'onActivate');
 
         const item1Element = overlayContainerElement.querySelector('#menu-item-1') as HTMLButtonElement;
         expect(item1Element).toBeTruthy();
         item1Element.dispatchEvent(new KeyboardEvent('keydown', { key: 'Enter' }));
 
-        expect(component.item1Activated).toHaveBeenCalledTimes(1);
+        expect(component.onActivate).toHaveBeenCalledTimes(1);
     });
+
+    it('should not emit a click or activated event when disabled is true.', async () => {
+        component.disabled = true;
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        // open menu
+        component.trigger.openMenu();
+        fixture.detectChanges();
+        await fixture.whenStable();
+
+        // get the child submenu item
+        const itemWithSubMenu = overlayContainerElement.querySelector('#menu-item-1') as HTMLButtonElement;
+        expect(itemWithSubMenu).toBeTruthy();
+
+        const subMenuItem = document.querySelector<HTMLButtonElement>(
+            '#menu-item-disabled'
+        );
+        expect(subMenuItem).toBeTruthy();
+
+        spyOn(component, 'onActivate');
+        spyOn(component, 'onClick');
+
+        // perform a click
+        itemWithSubMenu.click();
+        subMenuItem.click();
```

Comment:
When I suggested that it was okay to combine `click` and `activate` into the same test, I did not mean clicking separate items. You can and should bind both events to the same menu item.

The test should be:
1. Set disabled = true
2. Click the disabled menu item.
3. Verify that neither `activate` nor `click` were emitted.